File: regression/cs/tst11.cs
using Microsoft.Z3;
using System;

class Test
{

    public void Run()
    {        
        using (Context ctx = new Context())
        {
            RealExpr x = ctx.MkRealConst("x");
            RealExpr y = ctx.MkRealConst("y");

            Goal g = ctx.MkGoal();
            g.Assert(ctx.MkGt(x, ctx.MkReal(10)), ctx.MkEq(y, ctx.MkAdd(x, ctx.MkReal(1))));
            g.Assert(ctx.MkGt(y, ctx.MkReal(1)));

            Tactic t = ctx.MkTactic("simplify");
            ApplyResult r = t.Apply(g);
             
            Console.WriteLine(r);
            Console.WriteLine(g.Size);

            foreach (Goal s in r.Subgoals)
                Console.WriteLine(s);

            Console.WriteLine("Old goal: ");
            Console.WriteLine(g);

            t = ctx.Then(ctx.MkTactic("simplify"), ctx.MkTactic("solve-eqs"));
            r = t.Apply(g);

            Console.WriteLine(r);

            Solver solver = ctx.MkSolver();
            foreach (BoolExpr f in r.Subgoals[0].Formulas)
                solver.Assert(f);

            Console.WriteLine(solver);
            Console.WriteLine(solver.Check());
            Console.WriteLine(solver.Model);
            Console.WriteLine("applying model convert");
            Console.WriteLine(r.ConvertModel(0, solver.Model));
            Console.WriteLine("done");
        }
    }
}

File: old-regressions/python/tst11.py

# Copyright (c) 2015 Microsoft Corporation

from z3 import *

x = Real('x')
y = Real('y')
g = Goal()
g.add(x > 10, y == x + 1) 
g.add(y > 1)
t = Tactic('simplify')
r = t.apply(g)
print(r)
print(len(r))
for s in r:
    print(s)
print("old goal:")
print(g)
t = Then('simplify', 'solve-eqs')
r = t.apply(g)
print r
s = Solver()
for f in r[0]:
    s.add(f)
print(s)
print(s.check())
print(s.model())
print("applying model convert...")
print(r.convert_model(s.model()))
print("done")
